docs/topics/api/addons.rst

```diff
@@ -44,6 +44,10 @@ This endpoint allows you to search through public add-ons.
     :query string appversion: Filter by application version compatibility. Pass the full version as a string, e.g. ``46.0``. Only valid when the ``app`` parameter is also present.
     :query string author: Filter by exact author username. Multiple author names can be specified, separated by comma(s), in which case add-ons with at least one matching author are returned.
     :query string category: Filter by :ref:`category slug <category-list>`. ``app`` and ``type`` parameters need to be set, otherwise this parameter is ignored.
+    :query boolean featured: Filter to only featured add-ons.
+        If ``app`` is provided as a parameter then only featured collections targetted to that application are taken into account.
+        If ``lang`` is provided then only featured collections for that language, or no language, are taken into account. Both ``app`` and ``lang`` can be provided to filter to addons that are featured in that application and for that language, or no language.
+        The value `true` is optional - ``featured`` and ``featured=true`` are identical.
```

Comment:
> explicit is better than implicit

Either make that common among the API and document it once or always assume `featured=true` or `featured` (without `true`) imho. If we're doing that more often we should unify it and don't document it on every endpoint imho. Doesn't have to be fixed/done here and yet but should certainly be kept in mind.

Reply:
I'm not sure the best way to go.  I think we have other parameters passed in that could be thought as being xxx=True but we just test for the parameter's existence in the view code.  This is the first explicit Filter object written like that though.

Reply:
hmm, if we only test for it's existence then `featured=true` is misleading since `featured=false` doesn't do the opposite

Reply:
In _this_ case we test it properly and error on anything other than `true`, but I don't think that's the case everywhere in the API.  But I can change it so only `featured` is accepted with no value.

Reply:
I feel like that'd be cleaner. Summoning @diox here for opinions

Reply:
For consistency I think I'd prefer only supporting `featured=true`, and *not* `featured` without a value.

I can't think of any place in the API atm where the simple presence of the parameter is considered truthy. The closest type of parameter I can think of right now is `show_grouped_ratings` in the reviews API, and it does properly validate the values that are sent as booleans (and the documentation reflects this).

Reply:
https://github.com/mozilla/addons-server/pull/6339/commits/01ca3368eb01e1df1a0fc5277baf736f7eceecfc only accepts `featured=true`.  Plus adds more comments for glorious confusion free future.